pkg/kube/inject/inject.go

```diff
@@ -889,3 +898,61 @@ func GetProxyIDs(namespace *corev1.Namespace) (uid int64, gid int64) {
 
 	return
 }
+
+// GetProxyIDsFromPod returns the UID and GID to be used in the RunAsUser and RunAsGroup fields in the template
+// It relies on OpenShift injecting a valid UID into the pod template which we can then reuse.
+func GetProxyIDsFromPod(pod *corev1.Pod) (uid int64, gid int64) {
+	var proxyUID, proxyGID *int64
+	tproxyInterceptionMode := pod.Annotations != nil && pod.Annotations["sidecar.istio.io/interceptionMode"] == string(model.InterceptionTproxy)
+	if tproxyInterceptionMode {
+		proxyUID = ptr.To(int64(0))
+		proxyGID = ptr.To(constants.DefaultProxyUIDInt)
+	} else {
+		var err error
+		proxyUID, err = getProxyUIDFromAnnotation(*pod)
+		if err != nil {
+			log.Infof("Could not get proxyUID from annotation: %v", err)
+		}
+		if proxyUID == nil {
+			if pod.Spec.SecurityContext != nil && pod.Spec.SecurityContext.RunAsUser != nil {
+				proxyUID = ptr.To(*pod.Spec.SecurityContext.RunAsUser + 1)
```

Comment:
I know that this is the current state and we are unlikely to change it in 2.6, but this `+ 1` thing is fragile if the current value is the last in the range. I guess this is fine, plus hopefully 2.6 is the last release with it.

Reply:
yeah I kept thinking the same thing, but I don't think we should invest too much time in this now as it really is the last time we release it (and it's been fine so far)